evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -235,8 +235,9 @@ def test_multi_objective(X_y_multi):
     assert automl.problem_type == ProblemTypes.BINARY
 
 
-def test_categorical_classification(X_y_categorical_classification):
+def test_categorical_classification(AutoMLTestEnv, X_y_categorical_classification):
```

Comment:
By product of testing with `EvalMLAlgorithm`. I can remove this but I think these changes are good.